### ISSUE TITLE
Error when loading docx file generated by LibreOffice

### ISSUE THREAD

**Issue by taophp** [PHPOffice/PHPWord#889]
[UPDATE]: after studying the stack traces, I had to completely rewrite my issue. Sorry for any disagreement.[/UPDATE]
I have a docx file generated by LibreOffice. When I try to load it this way:

``` php
    $phpWord = \PhpOffice\PhpWord\IOFactory::load($this->templateFile);
```

And then I get this error:

``` php
PHP Fatal error:  Uncaught exception 'InvalidArgumentException' with message 'Invalid style value: none Options:bullet,decimal,upperRoman,lowerRoman,upperLetter,lowerLetter' in /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/AbstractStyle.php:289
Stack trace:
#0 /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/NumberingLevel.php(189): PhpOffice\PhpWord\Style\AbstractStyle->setEnumVal('none', Array, NULL)
#1 /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/AbstractStyle.php(166): PhpOffice\PhpWord\Style\NumberingLevel->setFormat('none')
#2 /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/AbstractStyle.php(181): PhpOffice\PhpWord\Style\AbstractStyle->setStyleValue('format', 'none')
#3 /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/Numbering.php(122): PhpOffice\PhpWord\Style\AbstractStyle->setStyleByArray(Array)
#4 /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/AbstractStyle.php(166): PhpOffice\PhpWord\Styl in /var/www/projects/ink2odt/vendor/phpoffice/phpword/src/PhpWord/Style/AbstractStyle.php on line 289
```

The origin of the problem seems to be that LibreOffice somehow uses the value `'none'` where PHPWord expects really nothing or `NULL`.

**Comment by Silencezjl** [comment]
How do you solve this problem? I have the same problem!

**Comment by taophp** [comment]
@Silencezjl Well, I did not. This was not a priority for me at the time... and it is even less now. But it seems not too hard to fix for someone who have a little time and some knowledge about PHP. Sorry I can't help more for the moment. Good luck !

**Comment by danielamogini** [comment]
Found a quick fix changing few lines from 288 to 292 PhpOffice\PhpWord\Style\AbstractStyle from:

`if ($value != null && trim($value) != '' && !empty($enum) && !in_array($value, $enum)) {`
            `throw new \InvalidArgumentException("Invalid style value: {$value} Options:".join(',', $enum));`
        `} elseif ($value === null || trim($value) == '') {`
            `$value = $default;`
        `}`

to

`if ($value != null && trim($value) != '' && $value != 'none' && !empty($enum) && !in_array($value, $enum)) {`
            `throw new \InvalidArgumentException("Invalid style value: {$value} Options:".join(',', $enum));`
        `} elseif ($value === null || trim($value || $value == 'none') == '') {`
            `$value = $default;`
        `}`

**Comment by troosan** [comment]
The latest development version supports all possible values of NumberFormat for the NumberingLevel format.
This was fixed in commit 8a5433e5df1f76d2bfa554b99e08e2bd79e05779
